crypto/zkp/schemes/cl/user_credential_receiver.go

```diff
@@ -137,24 +145,45 @@ func (r *UserCredentialReceiver) getCredentialRequestProoRandomfData(nymName str
 	if err != nil {
 		return nil, nil, fmt.Errorf("error when obtaining U proof random data: %v", err)
 	}
+
 	return nymProofRandomData, UProofRandomData, nil
 }
 
-func (r *UserCredentialReceiver) getCredentialRequestProofData(challenge *big.Int) ([]*big.Int, []*big.Int) {
-	return r.nymProver.GetProofData(challenge), r.UProver.GetProofData(challenge)
+func (r *UserCredentialReceiver) getCredentialRequestProofData(challenge *big.Int) ([]*big.Int, []*big.Int, error) {
+	return r.nymProver.GetProofData(challenge), r.UProver.GetProofData(challenge), nil
+}
+
+func (r *UserCredentialReceiver) getCommitmentsOfAttrsProof(challenge *big.Int) []*commitmentzkp.DFOpeningProof {
+	commitmentsOfAttrsProofs := make([]*commitmentzkp.DFOpeningProof, len(r.User.attrsCommitters))
+	for i, prover := range r.commitmentsOfAttrsProvers {
+		proofRandomData := prover.GetProofRandomData()
+		proofData1, proofData2 := prover.GetProofData(challenge)
+		commitmentsOfAttrsProofs[i] = commitmentzkp.NewDFOpeningProof(proofRandomData, challenge,
+			proofData1, proofData2)
+	}
+	return commitmentsOfAttrsProofs
 }
 
-func (r *UserCredentialReceiver) GetCredentialRequest(nymName string, nym, U, n1 *big.Int) (*dlogproofs.SchnorrProof,
-	*qrspecialrsaproofs.RepresentationProof, error) {
-	nymProofRandomData, UProofRandomData, err := r.getCredentialRequestProoRandomfData(nymName)
+// GetCredentialRequest computes U ...
+func (r *UserCredentialReceiver) GetCredentialRequest(nymName string, nym, n1 *big.Int) (*dlogproofs.SchnorrProof,
```

Comment:
1. See one of the other comments where I propose encapsulating nym value + name into a single struct called `Nym` or `NamedNym`. As it is now, we could just pass any nym name that does not correspond to the nym value at all, and it could lead to unexpected behavior.
2. From the test code, it is obvious that parameter `n1` (and in some other functions also `n2`) corresponds to nonce. For clarity, could you please **rename all nonce parameters - here and in other functions** - to `nonce`? It would add some meaning to the parameter, whereas `n1` and `n2` hold no meaning to an average Joe at all :slightly_smiling_face: